src/Nethermind/Nethermind.Tools.Kute/Application.cs

```diff
@@ -0,0 +1,83 @@
+// SPDX-FileCopyrightText: 2023 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using System.Diagnostics;
+using System.Text.Json;
+using Nethermind.Tools.Kute.JsonRpcMessageProvider;
+using Nethermind.Tools.Kute.JsonRpcMethodFilter;
+using Nethermind.Tools.Kute.JsonRpcSubmitter;
+using Nethermind.Tools.Kute.MetricsConsumer;
+
+namespace Nethermind.Tools.Kute;
+
+class Application
+{
+    private readonly Metrics _metrics = new();
+
+    private readonly IJsonRpcMessageProvider _msgProvider;
+    private readonly IJsonRpcSubmitter _submitter;
+    private readonly IMetricsConsumer _metricsConsumer;
+    private readonly IJsonRpcMethodFilter _methodFilter;
+
+    public Application(
+        IJsonRpcMessageProvider msgProvider,
+        IJsonRpcSubmitter submitter,
+        IMetricsConsumer metricsConsumer,
+        IJsonRpcMethodFilter methodFilter
+    )
+    {
+        _msgProvider = msgProvider;
+        _submitter = submitter;
+        _metricsConsumer = metricsConsumer;
+        _methodFilter = methodFilter;
+    }
+
+    public async Task Run()
+    {
+        var start = Stopwatch.GetTimestamp();
```

Comment:
Is this the best way to measure execution time in C#?

Reply:
Generally, yes.